source/MaterialXGraphEditor/Graph.h

```diff
@@ -246,8 +246,7 @@ class Graph
     std::vector<std::string> _currGraphName;
 
     // for adding new nodes
-    std::unordered_map<std::string, std::vector<mx::NodeDefPtr>> _nodesToAdd;
-    std::unordered_map<std::string, std::vector<std::vector<std::string>>> _extraNodes;
+    std::vector<std::vector<std::string>> _nodesToAdd;
```

Comment:
A niggly thing. This could be a vec of `StringVec` or could add a new new type `StringVecVec` :). 
?

Reply:
I've added a class `MenuItem` to clean this up